tests/base/src/test/java/org/keycloak/test/admin/authz/fgap/UserResourceTypePermissionTest.java

```diff
@@ -0,0 +1,196 @@
+            ScopePermissionResource permission = permissions.findById(permissionRep.getId());
+            assertEquals(expected.getName(), permissionRep.getName());
+            assertEquals(1, permission.scopes().size());
+            assertEquals(1, permission.resources().size());
+            assertEquals(3, permission.associatedPolicies().size());
+        }
+    }
+
+    @Test
+    public void testFindByResourceObject() {
+        ScopePermissionsResource permissions = client.admin().authorization().permissions().scope();
+
+        try (Response response = permissions.create(createUserPermission(userAlice))) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        }
+
+        try (Response response = permissions.create(createUserPermission(userBob))) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        }
+
+        List<ScopePermissionRepresentation> existing = permissions.findAll(null, null, userAlice.getId(), -1, -1);
+        assertEquals(1, existing.size());
+        existing = permissions.findAll(null, null, userBob.getId(), -1, -1);
+        assertEquals(1, existing.size());
+    }
+
+    @Test
+    public void testDelete() {
+        ScopePermissionsResource permissions = client.admin().authorization().permissions().scope();
+        ScopePermissionRepresentation expected = createUserPermission(userAlice);
+
+        try (Response response = permissions.create(expected)) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        }
+
+        expected  = createUserPermission(userBob);
+
+        try (Response response = permissions.create(expected)) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+        }
+
+        List<ScopePermissionRepresentation> existing = permissions.findAll(null, null, userAlice.getId(), -1, -1);
+        assertEquals(1, existing.size());
+        permissions.findById(existing.get(0).getId()).remove();
+        existing = permissions.findAll(null, null, userAlice.getId(), -1, -1);
+        assertEquals(0, existing.size());
+
+        existing = permissions.findAll(null, null, userBob.getId(), -1, -1);
+        assertEquals(1, existing.size());
+    }
+
+    private ScopePermissionRepresentation createUserPermission(ManagedUser user) {
+        ScopePermissionRepresentation permission = new ScopePermissionRepresentation();
+
+        permission.setName(KeycloakModelUtils.generateId());
+        permission.setResourceType(AdminPermissionsAuthorizationSchema.USERS.getType());
+        permission.setResources(Set.of(user.getUsername()));
+        permission.setScopes(AdminPermissionsAuthorizationSchema.USERS.getScopes());
+        permission.setPolicies(Set.of("User Policy 0", "User Policy 1", "User Policy 2"));
+
+        return permission;
+    }
+
+    private static class RealmAdminPermissionsConfig implements RealmConfig {
```

Comment:
Maybe move RealmAdminPermissionsConfig and KeycloakAdminPermissionsServerConfig to outer classes so they can be shared across multiple FGA tests?

Reply:
Yes, that is the idea once we have write more tests.